src/QueryCreator.tsx

```diff
@@ -113,19 +135,20 @@ class QueryCreator extends React.PureComponent<Props> {
         state.lastAction.type.includes(QUERY_UPDATE_ACTION)
       ) {
         if (this.updateQueryTrigger) clearTimeout(this.updateQueryTrigger);
+
         this.updateQueryTrigger = setTimeout(() => {
           onUpdateQuery(transformToQuery(query), isQueryReady);
         }, UPDATE_TIMEOUT);
       }
     });
   };
 
-  subscribeSetQuery = () => {
+  subscribeSetQuery = (defaultTimezoneForQuery: string) => {
     this.setQuerySubscription = this.pubsub.subscribe(
       (eventName: string, meta: any) => {
         switch (eventName) {
           case NEW_QUERY_EVENT:
-            this.store.dispatch(resetQuery());
+            this.store.dispatch(resetQuery(defaultTimezoneForQuery));
```

Comment:
As `<QueryCreator />` is react component and `defaultTimezoneForQuery` could potentially change - I suggest to avoid using a function argument closure and get default timezone directly from store before dispatching  `resetQuery()` action.